### ISSUE TITLE
Feature: Multi Pie "Autodomain"

### ISSUE THREAD

**Issue by ebellumat** [plotly/plotly.js#2909]
When we create dynamic pie charts, it is difficult to calculate the "domain" of each graph. If you had a property called "autodomain" for the pie charts to position themselves properly, or some other property that could be defined how many pie charts would be rendered on a line, it would be easier to use.

**Comment by alexcjohnson** [comment]
Right, like we do for multiple 3D scenes or multiple maps. Worth considering. That said, have you looked at `layout.grid`? If you use that, you can just specify `domain: {row, column}` for each pie trace rather than `domain: {x, y}` which, you're right, is awkward to calculate, but you get the flexibility to decide on side-by-side, top-to-bottom, or any shape of grid.

**Comment by alexcjohnson** [comment]
> can you create a jsFiddle with this `layout.grid` sample?

https://codepen.io/alexcjohnson/pen/NBZYpm?editors=0010

**Comment by ebellumat** [comment]
Oh my god! Much easier! This sample in the docs would be wonderful!

https://plot.ly/javascript/pie-charts/

**Comment by ebellumat** [comment]
Result in Pivottable JS after changes. I Create a Pie/Donut Chart render.

![image](https://user-images.githubusercontent.com/26942680/44224351-87ad1600-a160-11e8-9d9a-de606ad726c1.png)

![image](https://user-images.githubusercontent.com/26942680/44224362-90055100-a160-11e8-92ef-1dd143f241cd.png)

![image](https://user-images.githubusercontent.com/26942680/44224408-aad7c580-a160-11e8-87bb-5433d0d826d5.png)

**Comment by gvwilson** [comment]
Hi - this issue has been sitting for a while, so as part of our effort to tidy up our public repositories I'm going to close it. If it's still a concern, we'd be grateful if you could open a new issue (with a short reproducible example if appropriate) so that we can add it to our stack. Cheers - @gvwilson
